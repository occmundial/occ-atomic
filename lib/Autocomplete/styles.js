import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
