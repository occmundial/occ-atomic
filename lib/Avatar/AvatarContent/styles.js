import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
