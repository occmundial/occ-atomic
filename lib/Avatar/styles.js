import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
