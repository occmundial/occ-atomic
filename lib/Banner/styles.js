import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
