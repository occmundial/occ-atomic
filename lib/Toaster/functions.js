import isEmpty from 'lodash/isEmpty';

export const toaster = {
	toastId: 0,
	toast: null,
	success: function(value) { value.type = 'success'; this.addToast(value); },
	error: function(value) { value.type = 'error'; this.addToast(value); },
	info: function(value) { value.type = 'info'; this.addToast(value); },
	warning: function(value) { value.type = 'warning'; this.addToast(value); },
	promote: function(value) { value.type = 'promote'; this.addToast(value); },
	close: function() {
		if (this.closeListener && !isEmpty(this.toast)) this.closeListener(this.toast);
	},
	addToast: function(value) {
		if (this.addListener) {
			this.toastId++;
			this.toast = value;
			this.addListener(this.toast, this.toastId);
		}
	},
	removeToast: function() { this.toast = null; },
	addListener: null,
	registerAddListener: function(listener) { this.addListener = listener; },
	closeListener: null,
	registerCloseListener: function(listener) { this.closeListener = listener; }
};

export function Timer(callback, delay) {
    let timerId, start, remaining = delay;

    this.cancel = function() {
        clearTimeout(timerId);
    };

    this.pause = function() {
        window.clearTimeout(timerId);
        remaining -= new Date() - start;
    };

    this.resume = function() {
        start = new Date();
        window.clearTimeout(timerId);
        timerId = window.setTimeout(callback, remaining);
    };

    this.resume();
};
