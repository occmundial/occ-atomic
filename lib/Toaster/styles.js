import grid from '../subatomic/grid';
import spacing from '../tokens/spacing.json';

export default {
	container: {
		position: 'fixed',
		zIndex: 1001,
		bottom: 0,
		left: 0,
		padding: spacing['size-4'],
		[`@media screen and (min-width:${grid.xs}px)`]: {
			padding: spacing['size-6'],
		},
		[`@media screen and (max-width:${grid.xs-1}px)`]: {
			width: '100vw'
		},
		pointerEvents: 'none'
	}
};
