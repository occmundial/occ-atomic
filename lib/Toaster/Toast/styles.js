import colors from '../../tokens/colors.json';
import fonts from '../../tokens/fonts.json';
import spacing from '../../tokens/spacing.json';
import borderRadius from '../../tokens/borderRadius.json';
import shadows from '../../tokens/shadows.json';
import grid from '../../subatomic/grid';
import { parseFontValue } from '../../Text/styles';

export default {
	'@keyframes slideToasterIn': {
		from: { transform: 'translateX(-100%)', opacity: 0 },
		to: { transform: 'translateX(0)', opacity: 1 }
	},
	toast: {
		borderRadius: borderRadius['br-xs'],
		boxShadow: shadows['elevation-elevation-5'],
		padding: spacing['size-3'],
		pointerEvents: 'auto',
		zIndex: 1001,
		animation: 'slideToasterIn 0.3s ease-out',
		transition: '0.3s all ease-out',
    outlineOffset: '-1px',
		[`@media screen and (min-width:${grid.xs}px)`]: {
			maxWidth: 480,
			minWidth: 400,
			padding: spacing['size-4']
		},
		[`@media screen and (max-width:${grid.xs-1}px)`]: {
			width: '100%'
		}
	},
	closing: {
		transform: 'translateX(-100%)',
		opacity: 0
	},
	success: {
    backgroundColor: colors.alert.success.bg,
		outline: `1px solid ${colors.alert.success.border}`,
	},
	error: {
    backgroundColor: colors.alert.error.bg,
		outline: `1px solid ${colors.alert.error.border}`,
	},
	warning: {
    backgroundColor: colors.alert.warning.bg,
		outline: `1px solid ${colors.alert.warning.border}`,
	},
	info: {
    backgroundColor: colors.alert.info.bg,
		outline: `1px solid ${colors.alert.info.border}`,
	},
	promote: {
    backgroundColor: colors.alert.neutral.bg,
		outline: `1px solid ${colors.alert.neutral.border}`,
	},
	textInfo: {
    color: colors.text.indigo.primary,
	},
  textWarning: {
    color: colors.text.warning,
	},
  textSuccess: {
    color: colors.text.success,
	},
  textError: {
    color: colors.text.error,
	},
  textPromote: {
    color: colors.text.white.primary,
	},
	title: {
    font: parseFontValue(fonts['alert-default']),
		margin: 0,
	},
	description: {
    font: parseFontValue(fonts['alert-description']),
		margin: [spacing['size-1'], 0, 0],
	},
	content: {
		flexDirection: 'column',
		[`@media screen and (min-width:${grid.xs}px)`]: {
			flexDirection: 'row',
		}
	},
	icon: {
		marginRight: spacing['size-2'],
		alignSelf: 'start'
	},
	actionWrap: {
		height: '100%'
	},
	actionText: {
    font: parseFontValue(fonts['link-small-strong']),
		whiteSpace: 'nowrap',
    textDecoration: 'underline',
    cursor: 'pointer',
		alignSelf: 'start',
		margin: [spacing['size-2'], 0, 0],
		[`@media screen and (min-width:${grid.xs}px)`]: {
			margin: [0, 0, 0, spacing['size-4']],
			alignSelf: 'center',
		},
	},
	closeIcon: {
		marginLeft: spacing['size-4'],
		alignSelf: 'center',
	},
};
