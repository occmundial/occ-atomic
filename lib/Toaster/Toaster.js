import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import Toast from './Toast';
import { toaster, Timer } from './functions';

const timings = {
	normal: 5000,
	longer: 8000
};

/**
* The Toaster component allows you to trigger any of the predefined types of alerts. Insert the Toaster component in any part of your site, and handle the alerts with the toaster object's API.
* Check the example to see how to trigger any kind of alert. This component will only show one toast at a time, any new toast will automatically dismiss the current toast.
* The toast object can have the next properties:
* • title - string
* • description -string
* • hasIcon - boolean
* • action - object with label and onClick
* • timer - one of: 'normal', 'longer'
* You can only mount one Toaster component at a time, since mounting an instance sets the toaster listeners, and unmounting the component nullifies the existing listeners, so having two instances of Toaster and unmounting one will break them.
*/
class Toaster extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			toast: null,
			toastId: null
		};
	}

	componentDidMount() {
		toaster.registerAddListener(this.onAdd);
		toaster.registerCloseListener(this.onClose);
	}

	componentWillUnmount() {
		toaster.registerAddListener(null);
		toaster.registerCloseListener(null);
		this.resetTimer();
	}

	onAdd = (toast, toastId) => {
		this.resetTimer();
		toast.closing = false;
		this.setState({ toast, toastId }, () => {
			if (!toast.closeIcon)
				this.addTimer();
		});
	}

	onClose = toast => {
		this.resetTimer();
		toast.closing = true;
		this.setState({ toast });
		this.timer = new Timer(() => {
			toaster.removeToast();
			this.setState({ toast: null });
		}, 300);
	}

	pauseTimer = () => this.timer.pause();

	resumeTimer = () => this.timer.resume();

	addTimer = () => {
		const { toast } = this.state;
		const defaultTime = timings.normal;
		const time = toast.timer ? timings[toast.timer] || defaultTime : defaultTime;
		this.timer = new Timer(() => {
			this.onClose(toast);
		}, time);
	}

	resetTimer = () => { if (this.timer) this.timer.cancel(); }

	render() {
		const { toast, toastId } = this.state;
		const { classes, container } = this.props;
		const ToasterComponent = (
			<div className={classes.container}>
				{!!toast && (
					<Toast
						key={toastId}
						description={toast.description}
						title={toast.title}
						theme={toast.type}
						action={toast.action}
						closeIcon={toast.closeIcon}
						closing={toast.closing}
						testId={toast.testId}
						onClose={() => this.onClose(toast)}
						pauseTimer={this.pauseTimer}
						resumeTimer={this.resumeTimer}
					/>
				)}
			</div>
		);
        return container ? (
			ReactDOM.createPortal(
                ToasterComponent,
                container
            )
		) : ToasterComponent;
	}
}

Toaster.propTypes = {
	classes: PropTypes.object.isRequired,
	/** DOM Element to append the Toaster component into it. If you don't specify a container the Toaster will be rendered in the exact same DOM position you used it. This can in some cases generate problems with z-index or position. */
	container: PropTypes.object
};

export default Toaster;
