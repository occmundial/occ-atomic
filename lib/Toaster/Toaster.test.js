import React from 'react';
import { mount } from 'enzyme';
import Toaster from './Toaster';
import { toaster } from './functions';
import styles from './styles';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Toaster", () => {

    it('matches the snapshot', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('launches a toast', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        toaster.success({
            title: 'Title'
        });
        wrapper.update();
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('p').text()).toBe('Title');
    });

    it('removes a toast', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        toaster.success({
            title: 'Title'
        });
        wrapper.update();
        expect(wrapper.find('p').text()).toBe('Title');
        toaster.close();
        setTimeout(() => {
            wrapper.update();
            expect(wrapper.find('p').length).toBe(0);
        }, 500);
    });

    it('replaces the current toast', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        toaster.success({
            title: 'Title'
        });
        wrapper.update();
        expect(wrapper.find('p').text()).toBe('Title');
        toaster.success({
            title: 'New title'
        });
        wrapper.update();
        expect(wrapper.find('p').text()).toBe('New title');
    });

    it('removes a toast after the timer finishes', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        toaster.success({
            title: 'Title'
        });
        wrapper.update();
        expect(wrapper.find('p').text()).toBe('Title');
        setTimeout(() => {
            wrapper.update();
            expect(wrapper.find('p').length).toBe(0);
        }, 5000);
    });

    it('calls the onClick function', () => {
        const wrapper = mount(
            <Toaster classes={classes} />
        );
        const action = {
            label: 'Action',
            onClick: jest.fn()
        };
        toaster.success({
            title: 'Title',
            action,
            testId: "toaster-test"
        });
        wrapper.update();
        wrapper.find({ "data-testid": 'toaster-test__link'}).simulate('click');
        expect(action.onClick).toHaveBeenCalled();
        setTimeout(() => {
            wrapper.update();
            expect(wrapper.find('p').length).toBe(0);
        }, 0);
    });

});
