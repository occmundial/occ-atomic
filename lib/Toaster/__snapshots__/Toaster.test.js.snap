// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster launches a toast 1`] = `"<div class=\\"container\\"><div class=\\"Toast-toast-0-1-1 Toast-success-0-1-3\\"><div class=\\"Flexbox-flex-0-1-12 Toast-content-0-1-7\\"><div class=\\"Flexbox-flex-0-1-12 Flexbox-acenter-0-1-29\\"><div class=\\"\\"><p class=\\"Text-text-0-1-44 Text-subheading-0-1-48 Text-white-0-1-61\\">Title</p></div></div></div></div></div>"`;

exports[`Toaster matches the snapshot 1`] = `"<div class=\\"container\\"></div>"`;
